explorer/src/pages/TransactionDetailsPage.tsx

```diff
@@ -410,6 +415,129 @@ function AccountsCard({
   );
 }
 
+type TokenBalanceRow = {
+  account: PublicKey;
+  mint: string;
+  balance: TokenAmount;
+  delta: number;
+  accountIndex: number;
+};
+
+function generateTokenBalanceRows(
+  preTokenBalances: TokenBalance[],
+  postTokenBalances: TokenBalance[],
+  accounts: ParsedMessageAccount[]
+): TokenBalanceRow[] {
+  let preBalanceMap: { [index: number]: TokenBalance } = {};
+
+  preTokenBalances.forEach(
+    (balance) => (preBalanceMap[balance.accountIndex] = balance)
+  );
+
+  return postTokenBalances
+    .map(
+      ({ uiTokenAmount, accountIndex, mint }): TokenBalanceRow => {
+        let delta;
+
+        if (accountIndex in preBalanceMap) {
+          delta = normalizeTokenDelta(
+            uiTokenAmount,
+            preBalanceMap[accountIndex].uiTokenAmount
```

Comment:
Unlikely to happen but it's possible that the mint of a token changed during a transaction (it could be closed and re-initialized). Can you make sure that the mint matches before computing the delta?

Reply:
Sure thing. What would be the expected behavior in this kind of situation?

Reply:
Probably flatten into two separate rows, one for each mint

Reply:
Since we're reflecting the post balance of an account, maybe it would be better to just throw out the pre balance with the former mint? So a single row?

Reply:
Isn't the intent to show the delta? If I have balance A in token X to balance B in token Y I would expect to see a line item for a full withdrawal of A for token X and then a positive delta of B in token Y

Reply:
Yep -- that makes complete sense. I wasn't grokking that that could happen in one transaction. Thanks for the clarification.

Reply:
![Explorer-Solana (6)](https://user-images.githubusercontent.com/188792/106524193-1039e100-6497-11eb-8853-558185ba84e7.png)